gateway-server/src/main/java/org/apache/knox/gateway/session/control/InMemoryConcurrentSessionVerifier.java

```diff
@@ -122,17 +129,34 @@ public void init(GatewayConfig config, Map<String, String> options) throws Servi
     this.nonPrivilegedUsers = config.getNonPrivilegedUsers();
     this.privilegedUserConcurrentSessionLimit = config.getPrivilegedUsersConcurrentSessionLimit();
     this.nonPrivilegedUserConcurrentSessionLimit = config.getNonPrivilegedUsersConcurrentSessionLimit();
+    this.cleaningPeriod = config.getConcurrentSessionVerifierExpiredTokensCleaningPeriod();
     this.concurrentSessionCounter = new ConcurrentHashMap<>();
   }
 
   @Override
   public void start() throws ServiceLifecycleException {
-
+    expiredTokenRemover.scheduleAtFixedRate(this::removeExpiredTokens, cleaningPeriod, cleaningPeriod, TimeUnit.SECONDS);
   }
 
   @Override
   public void stop() throws ServiceLifecycleException {
+    expiredTokenRemover.shutdown();
+  }
 
+  private void removeExpiredTokens() {
+    sessionCountModifyLock.lock();
+    try {
+      Iterator<Map.Entry<String, Set<SessionJWT>>> concurrentSessionCounterIterator = concurrentSessionCounter.entrySet().iterator();
+      while (concurrentSessionCounterIterator.hasNext()) {
+        Map.Entry<String, Set<SessionJWT>> concurrentSessionCounterMapEntry = concurrentSessionCounterIterator.next();
+        concurrentSessionCounterMapEntry.setValue(concurrentSessionCounterMapEntry.getValue().stream().filter(sessionJWT -> !sessionJWT.hasExpired()).collect(Collectors.toSet()));
```

Comment:
This might be shorter and easer to read than the filter expression. Plus it's probably faster too.

```java
concurrentSessionCounterMapEntry.getValue().removeIf(session -> session.hasExpired());
```

Reply:
Yes, thats an easier way. Thank you.